src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -20,41 +20,49 @@ public sealed partial class JsonSerializerOptions
         private static readonly Dictionary<Type, JsonConverter> s_defaultSimpleConverters = GetDefaultSimpleConverters();
 
         // The global list of built-in converters that override CanConvert().
-        private static readonly List<JsonConverter> s_defaultFactoryConverters = GetDefaultConverters();
+        private static readonly JsonConverter[] s_defaultFactoryConverters = new JsonConverter[]
+        {
+            new JsonConverterEnum(),
+            new JsonKeyValuePairConverter()
+        };
 
         // The cached converters (custom or built-in).
         private readonly ConcurrentDictionary<Type, JsonConverter?> _converters = new ConcurrentDictionary<Type, JsonConverter?>();
 
         private static Dictionary<Type, JsonConverter> GetDefaultSimpleConverters()
         {
+            const int NumberOfSimpleConverters = 21;
             var converters = new Dictionary<Type, JsonConverter>(NumberOfSimpleConverters);
 
             // Use a dictionary for simple converters.
-            foreach (JsonConverter converter in DefaultSimpleConverters)
-            {
-                converters.Add(converter.TypeToConvert!, converter);
-            }
-
+            // When adding to this, update NumberOfSimpleConverters above.
```

Comment:
This comment isn't really necessary.  If someone adds to this and doesn't update the number, the assert will fire.

Reply:
You can probably write this as:

```csharp
return new Dictionary<Type, JsonConverter> {
    [JsonConverterBoolean.TypeToConvert!] = new JsonConverterBoolean(),
    [JsonConverterByte.TypeToConvert!] = new JsonConverterByte(),
    [JsonConverterByteArray.TypeToConvert!] = new JsonConverterByteArray(),
    [JsonConverterChar.TypeToConvert!] = new JsonConverterChar(),
    [JsonConverterDateTime.TypeToConvert!] = new JsonConverterDateTime(),
    [JsonConverterDateTimeOffset.TypeToConvert!] = new JsonConverterDateTimeOffset(),
    [JsonConverterDouble.TypeToConvert!] = new JsonConverterDouble(),
    [JsonConverterDecimal.TypeToConvert!] = new JsonConverterDecimal(),
    [JsonConverterGuid.TypeToConvert!] = new JsonConverterGuid(),
    [JsonConverterInt16.TypeToConvert!] = new JsonConverterInt16(),
    [JsonConverterInt32.TypeToConvert!] = new JsonConverterInt32(),
    [JsonConverterInt64.TypeToConvert!] = new JsonConverterInt64(),
    [JsonConverterJsonElement.TypeToConvert!] = new JsonConverterJsonElement(),
    [JsonConverterObject.TypeToConvert!] = new JsonConverterObject(),
    [JsonConverterSByte.TypeToConvert!] = new JsonConverterSByte(),
    [JsonConverterSingle.TypeToConvert!] = new JsonConverterSingle(),
    [JsonConverterString.TypeToConvert!] = new JsonConverterString(),
    [JsonConverterUInt16.TypeToConvert!] = new JsonConverterUInt16(),
    [JsonConverterUInt32.TypeToConvert!] = new JsonConverterUInt32(),
    [JsonConverterUInt64.TypeToConvert!] = new JsonConverterUInt64(),
    [JsonConverterUri.TypeToConvert!] = new JsonConverterUri(),
};
```

It's a little repetitive, yes, but then you get rid of the assertion, NumberOfSimpleConverters, and the comment.

Reply:
+1

Reply:
The original reason I added is so you don't have to run a test to figure it out. Anyway it can be removed.